File: app/src/main/java/com/example/smishingdetectionapp/ApiService.java
package com.example.smishingdetectionapp;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface ApiService {
    @POST("/submitReport")
    Call<ReportResponse> submitReport(@Body Report report);

    @POST("/submitFeedback")
    Call<ReportResponse> submitFeedback(@Body Report report);
}

File: app/src/main/java/com/example/smishingdetectionapp/net/models/ApiService.kt
package com.example.smishingdetectionapp.net

import com.example.smishingdetectionapp.net.models.ReportRequest
import com.example.smishingdetectionapp.net.models.ReportResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface ApiService {
    @POST("api/reports")
    suspend fun postReport(@Body body: ReportRequest): Response<ReportResponse>
}
